spf/dnscache.go

```diff
@@ -0,0 +1,26 @@
+package spf
+
+type dnsCache map[string]map[string][]string
```

Comment:
We should document the dimensions here. Looks like it is [name][rType] -> answers. Some questions:

1. Do we intend to ever do A/MX resolving, or just TXT? If just TXT, we can probably eliminate the record type dimension.
2. Why store all the TXT responses? If we filter to those with prefix `v=spf1` there should be 0 or 1. This could be as simple as `map[string]string`.